gamestonk_terminal/portfolio/dg_api.py

```diff
@@ -0,0 +1,59 @@
+import degiroapi
+from termcolor import colored
+from gamestonk_terminal.config_terminal import DG_USERNAME as user, DG_PASSWORD as pw
+
+degiro = degiroapi.DeGiro()
+
+
+def login():
+    while True:
+        login_prefr = input("Do you have 2FA activated? - y/n> ")
+        if login_prefr == "q":
+            break
+        elif login_prefr == "y":
+            topt = input("Enter TOPT> ")
+            degiro.login(user, pw, topt)
+            break
+        elif login_prefr == "n":
+            degiro.login(user, pw)
+            break
```

Comment:
Good point, in that case I'll keep the regular logging and I can have a look at adding the 2FA option in the port_controller instead in the future. So that there is an optional flag for choosing 2FA logging otherwise it uses the normal one -> login --totp degiro, or something like that. Do you think that would work?

> And also do `poetry add degiroapi` as well as add it to the requirements files.
Sorry for the noob question, but is this just doing pip freeze > requirements.txt or is there a different way of doing it?



> Another question: can you upload an example of what the screenshot looks like after this is run? As for your comment about merging : is `breakeven` only for options? That would seem like the cost basis to me if that exists in the stocks

![image](https://user-images.githubusercontent.com/42816450/115779989-8b34d880-a3b0-11eb-8bd9-934266bffc97.png)

The break even option is available for stocks so that makes sense in that case - nice catch, haven't checked with options yet actually. Been a while since I got burned on that but I'll check it out tomorrow.

Reply:
> Do you think that would work? This would also be up to you guys, do you plan on supporting 2FA for other brokers in the future? 

I think that would work.  Also as for TFA, I guess it would depend on the user and also on how the API deals with tfa.  I think with RH, you get a text/call as the tfa, so no need to do anything in the terminal

> Sorry for the noob question, but is this just doing pip freeze > requirements.txt or is there a different way of doing it?
So just figure out what version diegroapi is (looks like 0.9.5).  Then in the requirements.txt files (I guess all of them), you would just make a new line
`diegroapi==0.9.5`

> The break even option is available for stocks so that makes sense in that case - nice catch, haven't checked with options yet actually. Been a while since I got burned on that but I'll check it out tomorrow.

Thanks for the picture.  Looks nice.  I would drop the ProductType, as I only consider stocks in the others, no options or crypto.  I also think that since the breakeven is the "cost basis", you could add that to the merge (if the columns align -- if they don't you could always make the data frame have a "" column...)

Reply:
You can use poetry to update requirements.txt file, with
```
poetry export -f requirements.txt --output requirements.txt --without-hashes
```

Reply:
> I think that would work. Also as for TFA, I guess it would depend on the user and also on how the API deals with tfa. I think with RH, you get a text/call as the tfa, so no need to do anything in the terminal

Ahh I see, with Degiro you scan a QR code and then use Google Auth. I think I'll hold off on that option for this PR and I'll put in a different PR for it. Was thinking of a way to handle TFA in the terminal automatically by having the user submit the secretkey they get from the QR code and generate the topt through that...

> So just figure out what version diegroapi is (looks like 0.9.5). Then in the requirements.txt files (I guess all of them), you would just make a new line
> `diegroapi==0.9.5`

Thanks, I'll try this or @DidierRLopes suggestion below.


> Thanks for the picture. Looks nice. I would drop the ProductType, as I only consider stocks in the others, no options or crypto. I also think that since the breakeven is the "cost basis", you could add that to the merge (if the columns align -- if they don't you could always make the data frame have a "" column...)

Thanks for the feedback, glad you like it. Cool, I'll do the same in that case. Do you happen to have a screenshot of how the merge portfolio command looks like when executed?